### PR TITLE
[webgpu] Add binary WGSL support

### PR DESCRIPTION
**Pull request by axinging**
To test WGSL, change WEBGPU_USE_GLSL in tfjs-backend-webgpu/src/flags_webgpu.ts to false.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5322)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @haoyunfeix @xhcao @gyagp, PTAL

**Comment by qjia7** [comment]
Please make sure all yarn test are `PASS`. I met 9 failures with `WEBGPU_USE_GLSL` as `false`.

File: tfjs-backend-webgpu/src/kernels/binary_op_util.ts
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/binary_op_util.ts:140]: If https://github.com/gpuweb/gpuweb/issues/1910 is supported by chrome canary, we can use the same code snippet for `EQUAL` between glsl and wgsl? Please also check other places.

File: tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts
Comment (qjia7) [inline tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts:229]: Please add a comment to point to the bug you reported and state that it is a workaround. Similar for other places.

File: tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts
Comment (qjia7) [inline tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts:43]: Why we need to change it from `bool` to `i32`?

Comment (axinging) [reply tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts:43]: 'vec4<bool>' cannot be used in storage class 'storage' as it is non-host-shareable

File: tfjs-backend-webgpu/src/kernels/binary_op_util.ts
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/binary_op_util.ts]: Related workaround removed, PTAL.

File: tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts
Comment (axinging) [reply tfjs-backend-webgpu/src/shader_preprocessor_wgsl.ts]: Related workaround removed, PTAL.

**Review by qjia7 — APPROVED** [review]
@kainino0x @lina128 @jinjingforever Please take a look, thanks.

**Review by lina128 — APPROVED** [review]
LGTM from TFJS, please wait for @qjia7 and @kainino0x approval.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5322#-:-MgfqvCVA9AsnvU8TZjJ:bctgowr)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @jinjingforever, @kainino0x, and @qjia7)

**Review by kainino0x — APPROVED** [review]
Only had time to skim, but lgtm

**Pull request merged**
